.github/workflows/test-go-generate.yml

```
@@ -0,0 +1,40 @@
+name: Go generate check
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - "release/**"
+  pull_request:
+
+jobs:
+  go-generate:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+        with:
+          path: constellation
+          ref: ${{ github.head_ref }}
+
+      - name: Setup Go
+        uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+        with:
+          go-version: "1.19.3"
+          cache: true
+          cache-dependency-path: constellation/go.mod
+
+      - name: Install docgen
+        uses: ./constellation/.github/actions/install_docgen
+
+      - name: Install stringer
+        shell: bash
+        run: go install golang.org/x/tools/cmd/stringer@latest
+
+      - name: Go generate check
+        shell: bash
+        working-directory: constellation
+        run: |
+          go generate ./...
+          git diff --exit-code
```

This should be done with multi-module support.
```suggestion
      - name: Get Go submodules
        id: submods
        shell: bash
        run: |
          mods=$(go list -f '{{.Dir}}' -m | xargs)
          echo "Found mods: $mods"
          echo "submods=${mods}" >> "$GITHUB_OUTPUT"
       
      - name: Go generate check
        shell: bash
        working-directory: constellation
        run: |
          mods=${{ steps.submods.outputs.submods }}
          result=0
          for mod in $mods; do
              (cd $mod; go generate ./...; git diff --exit-code || result=1)
          done
          exit $result
```

---

Note for myself to check how this behaves if a tool referenced in go generate is not installed...